src/histolab/data/__init__.py

```diff
@@ -0,0 +1,264 @@
+# encoding: utf-8
+
+# ================This module is completely inspired by scikit-image================
+# https://github.com/scikit-image/scikit-image/blob/master/skimage/data/__init__.py
+# ==================================================================================
+
+import os
+import shutil
+import openslide
+
+from .. import __version__
+from ._registry import legacy_registry, registry, registry_urls
+
+legacy_data_dir = os.path.abspath(os.path.dirname(__file__))
+histolab_distribution_dir = os.path.join(legacy_data_dir, "..")
+
+try:
+    from pooch.utils import file_hash
+except ModuleNotFoundError:
+    # Function taken from
+    # https://github.com/fatiando/pooch/blob/master/pooch/utils.py
+    def file_hash(fname, alg="sha256"):
+        """
+        Calculate the hash of a given file.
+        Useful for checking if a file has changed or been corrupted.
+        Parameters
+        ----------
+        fname : str
+            The name of the file.
+        alg : str
+            The type of the hashing algorithm
+        Returns
+        -------
+        hash : str
+            The hash of the file.
+        Examples
+        --------
+        >>> fname = "test-file-for-hash.txt"
+        >>> with open(fname, "w") as f:
+        ...     __ = f.write("content of the file")
+        >>> print(file_hash(fname))
+        0fc74468e6a9a829f103d069aeb2bb4f8646bad58bf146bb0e3379b759ec4a00
+        >>> import os
+        >>> os.remove(fname)
+        """
+        import hashlib
+
+        if alg not in hashlib.algorithms_available:
+            raise ValueError("Algorithm '{}' not available in hashlib".format(alg))
+        # Calculate the hash in chunks to avoid overloading the memory
+        chunksize = 65536
+        hasher = hashlib.new(alg)
+        with open(fname, "rb") as fin:
+            buff = fin.read(chunksize)
+            while buff:
+                hasher.update(buff)
+                buff = fin.read(chunksize)
+        return hasher.hexdigest()
+
+
+def create_image_fetcher():
```

Comment:
this function is now private and i can't use type annotations here because pooch is imported in the try except statement. So the return values are pooch.Pooch obj + str, but pooch isn't imported by default so .... i would leave it as is